pkg/services/ngalert/api/api_ruler.go

```diff
@@ -330,6 +331,30 @@ func (srv RulerSrv) RouteGetRuleByUID(c *contextmodel.ReqContext, ruleUID string
 	return response.JSON(http.StatusOK, result)
 }
 
+func (srv RulerSrv) RouteGetRuleVersionsByUID(c *contextmodel.ReqContext, ruleUID string) response.Response {
+	ctx := c.Req.Context()
+	// make sure the user has access to the current version of the rule. Also, check if it exists
+	_, err := srv.getAuthorizedRuleByUid(ctx, c, ruleUID)
+	if err != nil {
+		if errors.Is(err, ngmodels.ErrAlertRuleNotFound) {
+			return response.Empty(http.StatusNotFound)
+		}
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to get rule by UID", err)
+	}
+
+	rules, err := srv.store.GetAlertRuleVersions(ctx, ngmodels.AlertRuleKey{OrgID: c.OrgID, UID: ruleUID})
+	if err != nil {
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to get rule history", err)
+	}
+	sort.Slice(rules, func(i, j int) bool { return rules[i].ID > rules[j].ID })
```

Comment:
We could probably order it at the database level? It'd be also easier to apply pagination in the future if needed

Reply:
Yea, I already sort it in the query https://github.com/grafana/grafana/blob/b95051c4afa7829e32f6d2ec8d27bf84f4b8a849/pkg/services/ngalert/store/alert_rule.go#L124. This one just makes sure that it's sorted in case we change the storage layer (and the tests)